### PR TITLE
Download script

### LINKED ISSUE #2: Explore fetching data from plant phenology database
A really useful resource for phenology data is http://plantphenology.org/.

The data are also available through a REST API (https://github.com/biocodellc/biscicol-server) and an R package is available to build a request: https://github.com/ropensci/rppo/blob/master/R/ppo_data.R. See documentation here: https://docs.ropensci.org/rppo/reference/rppo-package.html

It would be nice to see if we can build a similar, simple python script that uses this REST API to download data (ideally pep725) for a (small) region in Europe, for selected parameters, etc.

### PR DESCRIPTION
**Pull request by Peter9192**
This PR adds a python module to download data from the plant phenology data portal, very similar to [rppo](https://docs.ropensci.org/rppo/reference/rppo-package.html).

It can be used like so:

```py
from py_ppo import download

df = download(
    genus="Syringa",
    source="PEP725",
    year="[2000 TO 2021]",
    latitude="[40 TO 70]",
    longitude="[-10 TO 40]",
    termID="obo:PPO_0002313",
    explode=False,
    limit=1535,
    )
```

I also included an example script, which calculates the leafing-out dates based on the results from the above query, and plots it them on a map. This requires matplotlib and cartopy to be installed in your environment.

test_image

There's many remaining TODO's, but those can be addressed in follow-up PRs.

closes #2

### CODE REVIEW

File: py_ppo.py
Comment (SarahAlidoost) [inline py_ppo.py:78]:
```suggestion
        url = "https://biscicol.org/api/v2/ppo/all_short"
    elif present is True:
        url = "https://biscicol.org/api/v2/ppo/present_short"
    elif present is False:
        url = "https://biscicol.org/api/v2/ppo/absent_short"
```

File: py_ppo.py
Comment (SarahAlidoost) [inline py_ppo.py:82]: Lets add a `timeout=5 or 10` just in case :)

File: py_ppo.py
Comment (SarahAlidoost) [inline py_ppo.py:151]: why not pass a `downlod_dir` instead of using `TemporaryDirectory` as it generates random temp directories? I found that those random dirs are not accessible to users. As an user I want to check the content of the downloaded files and `data.csv`.

**Review by SarahAlidoost — CHANGES_REQUESTED** [review]
@Peter9192 nice work :+1:  after installing the python packages, I was able to run the example and tests. I left some comments, the major one is about the download directory, see my comment on the code. 

Also, the link that is returned to the terminal starts with http://purl.obolibrary.org which is linked to https://ontobee.org/ (the README.txt contains the information). I understand that this is the link to the raw dataset, but the link in the output shouldn't start with https://biscicol.org ? 

I think the files can be structured in sub-directories. But this can be done using a python template later.

File: test_download.py
Comment (SarahAlidoost) [inline test_download.py:33]:
```suggestion
        download(genus="Quercus")
```

**Comment by Peter9192** [comment]
> @Peter9192 nice work 👍 after installing the python packages, I was able to run the example and tests. 

Thanks for the review! 

> I left some comments, the major one is about the download directory, see my comment on the code.

I think this warrants a bit more discussion, so I've opened an issue: 

> Also, the link that is returned to the terminal starts with http://purl.obolibrary.org which is linked to https://ontobee.org/ (the README.txt contains the information). I understand that this is the link to the raw dataset, but the link in the output shouldn't start with https://biscicol.org ?

Hmm I agree it's not so nice. The terminal output is actually from the README that's retrieved from the server, so we can't change the text. Perhaps we should just print our own text here instead. That might also avoid confusion about the multiple output files.

> I think the files can be structured in sub-directories. But this can be done using a python template later.

Agree

File: py_ppo.py
Comment (Peter9192) [reply py_ppo.py:151]: Actually I didn't want the user to see that there are downloaded files and directories at all. I want to pretend the data came directly from the server. Perhaps with an updated output message you don't feel like you're missing out anymore? But see #7 for follow-up.

**Comment by Peter9192** [comment]
I updated the output message and opened #7. Hope this sufficiently addresses your concerns :-)

**Review by SarahAlidoost — APPROVED** [review]
@Peter9192 thanks for addressing my comments. Let's wait for @khzadeh 's review before merging the pull request.

File: py_ppo.py
Comment (khzadeh) [inline py_ppo.py:28]: My previous comment is on the todo list :) Here, I would add accepting a single coordinate with a buffer size too.

**Review by khzadeh — APPROVED** [review]
My comments are more related to future improvements. So I think this pull request is ready to be merged, thanks.

**Comment by Peter9192** [comment]
Thanks for the reviews!

**Pull request merged**
